modules/aql/arrow-aql/src/main/kotlin/arrow/aql/Min.kt

```diff
@@ -0,0 +1,31 @@
+package arrow.aql
+
+import arrow.core.ForId
+import arrow.core.Id
+import arrow.core.identity
+import arrow.core.value
+import arrow.core.Option
+import arrow.core.Some
+import arrow.core.None
+import arrow.typeclasses.Foldable
+import arrow.typeclasses.Order
+
+interface Min<F> {
+
+  fun foldable(): Foldable<F>
+
+  fun <A, Y, Z> Query<F, A, Y>.min(ord: Order<Z>, f: A.() -> Z): Query<ForId, Option<Z>, Option<Z>> =
```

Comment:
I'm still trying to figure out what happens with the `Y` lol

Reply:
I guess we take the result of the existing one and ignore it, so `Any` or `*` would work the same.